src/reusableState/customList/StringListReducers.ts

```diff
@@ -0,0 +1,60 @@
+import * as _ from 'underscore';
+import {IItemBoxProps} from '../../components/itemBox/ItemBox';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {IStringListPayload, StringListActions} from './StringListActions';
+
+export type IStringListCompositeState = {[id: string]: IStringListState};
+
+export interface IStringListState {
+    id: string;
+    list: string[];
+}
+
+export const stringListInitialState: IStringListCompositeState = {};
+
+export const convertStringListToItemsBox = (list: string[], itemsBoxParams: Partial<IItemBoxProps> = {}): IItemBoxProps[] => _.map(list, (value: string) => ({...itemsBoxParams, value}));
+export const convertItemsBoxToStringList = (items: IItemBoxProps[]): string[] => _.pluck(items, 'value');
+
+export const stringListCompositeState = (
+    state: IStringListCompositeState = stringListInitialState,
+    action: IReduxAction<IStringListPayload>,
+): IStringListCompositeState => {
+
+    if (!action.payload || (!state[action.payload.id] && action.type !== StringListActions.add)) {
+        return {...state};

```

Comment:
you can just return the state you don't need to destructure it since you make no modifications to it.